File: sum_of_primes.pl
#!/usr/bin/env perl

use Math::Complex;

sub prime {
    my $x = shift;
    my $s = sqrt($x);

    return 1 if $x == 2;

    for ( 2 .. ( $s + 1 ) ) {
        return 0 if $x % 2 == 0;
        return 0 if $x % $_ == 0;
    }

    return 1;
}

my $i = 2;
my $count = 0;
my $sum = 0;
while ( $count < 1000 ) {
    if ( prime($i) ) {
        $sum += $i;
        $count += 1;
    }
    $i += 1;
} 

print "$sum\n";

File: sum_of_primes.java

class Main {

    public static boolean isPrime(int x) {
        double sqr = Math.sqrt(x);

        for(int i = 2; i <= sqr; i++) {
           if(x % i == 0) {
               return false;
           }
        }

        return true;
    }

    public static void main(String[] args) {
        int sum = 0;
        int count = 0;
        int i = 2;

        while(count < 1000) {
            if(isPrime(i)) {
                sum += i;
                count++;
            }

            i++;
        }

        System.out.println(sum);
    }
}
